src/components/Carousel/index.tsx

```diff
@@ -0,0 +1,91 @@
+        if (wrapperRef.current) {
+            const itemWidth = (wrapperRef.current.clientWidth / children.length);
+            const maxIndex = (wrapperRef.current.scrollWidth - wrapperRef.current.clientWidth);
+            setCurrentIndex((prevIndex) => Math.min(prevIndex + itemWidth + scrollSpeed, maxIndex));
+        }
+    };
+
+    const handleMouseDown = (e: React.MouseEvent) => {
+        setDragStartX(e.clientX);
+        setDragging(true); // Allow dragging when move cursor
+    };
+
+    const handleMouseMove = (e: React.MouseEvent) => {
+        if (!dragging || dragStartX === null) return;
+        const dragDistance = (e.clientX - dragStartX) / 10;
+        setDragX(dragDistance);
+        if (wrapperRef.current) {
+            wrapperRef.current.style.transition = 'none';
+            wrapperRef.current.style.transform = `translateX(calc(-${(currentIndex / 10)}rem + ${dragDistance}rem))`;
+        }
+    };
+
+    const handleMouseUp = () => {
+        if (!dragging) return;
+        setDragging(false);
+        setDragStartX(null);
+        if (wrapperRef.current) {
+            const newIndex = currentIndex - (dragX * 10); // *10 convert from rem to px 
+            const maxIndex = (wrapperRef.current.scrollWidth - wrapperRef.current.clientWidth);
+            const clampedIndex = Math.max(0, Math.min(newIndex, maxIndex));
+            setCurrentIndex(clampedIndex);
+            setDragX(0);
+            wrapperRef.current.style.transition = 'transform 0.3s ease-in-out';
+            wrapperRef.current.style.transform = `translateX(-${(clampedIndex / 10)}rem)`;
+        }
+    };
+
+    return (
+        <CarouselContainer customWidth={customWidth}>
+            <StyledLeftArrow onClick={handlePrev} disabled={(currentIndex) === 0} />
+            <CarouselOverflow
+                onPointerDown={handleMouseDown}
+                onPointerMove={handleMouseMove}
+                onPointerUp={handleMouseUp}
+                onPointerLeave={handleMouseUp}
+            >
+                <CarouselWrapper
+                    ref={wrapperRef}
+                    // Updates the position of the carousel when the currentIndex changes, avoiding problems when the user uses the keyboard to navigate
+                    style={{ transform: `translateX(calc(-${(currentIndex / 10)}rem + ${dragX}rem))` }}
+                >
+                    {children.map((child) => {
+                        const key = (child as React.ReactElement).key;
+                        return (
+                        <CarouselItem 
+                            key={key ?? null}
+                        >
+                            {child}
+                        </CarouselItem>);
+                    })}
+                </CarouselWrapper>
+            </CarouselOverflow>
+            <StyledRightArrow onClick={handleNext} disabled={currentIndex >= ((wrapperRef.current?.scrollWidth) || 0) - (wrapperRef.current?.clientWidth || 0)} />
```

Comment:
onClick should probably not fire when disabled. I'm also [seeing a small bug](https://openstax.github.io/ui-components/refs/heads/CORE-675-filter-sections-component/250f84984d4cd6c08a351a31865c62c5e8d592fb/?story=carousel--default) with this where the initial load shows the right arrow as disabled when it shouldn't (it then renders correctly after going right and then back.)